### PR TITLE
refactor init

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description

- **fix: disable hyperlink functionality for unsupported shells**
- **refactor(color): move background Template to colors**
- **refactor(init): enable features automatically**
- **fix(zsh): correct logic to enable tooltips**
- **feat(cygwin): better detection**
- **refactor: better use of POSH_THEME**
- **feat(cache): split session and device cache**
- **feat(debug): provide shell name on debug**
- **feat(init): add --debug flag**
- **fix(windows): only reformat pwd on cygwin**
- **fix(fish): correctly export environment variables**

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/shell/scripts/omp.bash [hunk 1]
```diff
@@ -3,23 +3,30 @@ export POSH_SHELL_VERSION=$BASH_VERSION
...
+
+# switches to enable/disable features
+_omp_cursor_positioning=0
+_omp_ftcs_marks=0
 
 # start timer on command start
 PS0='${_omp_start_time:0:$((_omp_start_time="$(_omp_start_timer)",0))}$(_omp_ftcs_command_start)'
 # set secondary prompt
-PS2="$(::OMP:: print secondary --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION")"
+PS2="$(${_omp_executable} print secondary --shell=bash --shell-version="$BASH_VERSION")"
```
Comment (lewis-yeung) [inline src/shell/scripts/omp.bash:24]: This should be quoted to prevent word splitting after the variable expansion.

Comment (JanDeDobbeleer) [reply src/shell/scripts/omp.bash:24]: @lewis-yeung can you provide a quick PR? Maybe the expansion isn't needed here.

File: src/shell/scripts/omp.bash [hunk 2]
```diff
@@ -37,11 +44,11 @@ function _omp_set_cursor_position() {
 }
 
 function _omp_start_timer() {
-    ::OMP:: get millis
+    $_omp_executable get millis
 }
 
 function _omp_ftcs_command_start() {
-    if [[ "::FTCS_MARKS::" == "true" ]]; then
+    if [[ $_omp_ftcs_marks == 0 ]]; then
```
Comment (lewis-yeung) [inline src/shell/scripts/omp.bash:51]: It should be compared with `1` here.

File: src/shell/scripts/omp.bash [hunk 3]
```diff
@@ -75,19 +82,11 @@ function _omp_hook() {
...
     _omp_set_cursor_position
 
-    PS1="$(::OMP:: print primary --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --no-status="$_omp_no_exit_code" --terminal-width="${COLUMNS-0}" | tr -d '\0')"
+    PS1="$(${_omp_executable} print primary --shell=bash --shell-version="$BASH_VERSION" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --no-status="$_omp_no_exit_code" --terminal-width="${COLUMNS-0}" | tr -d '\0')"
 
     return $_omp_status_cache
 }
 
-if [[ "$TERM" != "linux" ]] && [[ -x "$(command -v ::OMP::)" ]] && ! [[ "$PROMPT_COMMAND" =~ "_omp_hook" ]]; then
+if [[ "$TERM" != "linux" ]] && [[ -x "$(command -v $_omp_executable)" ]] && ! [[ "$PROMPT_COMMAND" =~ "_omp_hook" ]]; then
```
Comment (lewis-yeung) [inline src/shell/scripts/omp.bash:90]: The check `[[ -x "$(command -v $_omp_executable)" ]]` is not needed.

File: src/shell/scripts/omp.fish [hunk 4]
```diff
@@ -1,12 +1,17 @@
-set --export POSH_THEME ::CONFIG::
-set --export POSH_SHELL_VERSION $FISH_VERSION
-set --global POWERLINE_COMMAND oh-my-posh
-set --global POSH_PID $fish_pid
-set --global CONDA_PROMPT_MODIFIER false
+set --global --export POSH_THEME ::CONFIG::
+set --global --export POSH_SHELL_VERSION $FISH_VERSION
+set --global --export POWERLINE_COMMAND oh-my-posh
+set --global --export POSH_PID $fish_pid
+set --global --export CONDA_PROMPT_MODIFIER false
```
Comment (lewis-yeung) [inline src/shell/scripts/omp.fish:5]: Here we don't have to specify `--global`, just `--export` is enough.

File: src/shell/scripts/omp.zsh [hunk 5]
```diff
@@ -184,22 +190,15 @@ function _omp_create_widget() {
...
 
-# legacy functions
-function enable_poshtooltips() {}
-function enable_poshtransientprompt() {}
-
-if [[ "::TOOLTIPS::" = "true" ]]; then
-  _omp_create_widget self-insert _omp_render_tooltip
-fi
+function enable_poshtooltips() {
+  widget=${$(bindkey " "):2}
```
Comment (lewis-yeung) [inline src/shell/scripts/omp.zsh:194]: We should use a local variable `widget` here.

File: src/shell/scripts/omp.bash (hunk 1 above)
Comment (lewis-yeung) [reply src/shell/scripts/omp.bash:24]: Sure. In a new PR, I'm going to include other changes that I suggested.

**Pull request merged**